Data/models/DataController.swift

```diff
@@ -37,6 +37,12 @@ public class DataController: NSObject {
         return container
     }()
     
+    private lazy var operationQueue: OperationQueue = {
+        let queue = OperationQueue()
+        queue.maxConcurrentOperationCount = 1
```

Comment:
This should not be a bottleneck even for heavy operations. We can always use operations with a different priority, so more important tasks will be finished first.
From my testing the queue count never exceeded 2 items, this will probably increase after implementing Sync.